perllib/FixMyStreet/Cobrand/Peterborough.pm

```diff
@@ -733,6 +737,100 @@ sub bulky_per_item_costs {
     return $cfg->{per_item_costs};
 }
 
+sub bulky_can_cancel {
+    my $self = shift;
+    my $c    = $self->{c};
+
+    my $pending_collection = $c->stash->{property}{pending_bulky_collection};
+
+    return
+           $pending_collection->external_id
+        && $self->bulky_can_view_collection($pending_collection)
+        && $self->within_bulky_cancel_window;
+}
+
+# Collections are scheduled to begin at 06:45 each day.
+# A cancellation made less than 24 hours before the collection is scheduled to
+# begin is not entitled to a refund.
+sub within_bulky_refund_window {
+    my $self = shift;
+    my $c    = $self->{c};
+
+    my $open_collection = $c->stash->{property}{pending_bulky_collection};
+    return 0 unless $open_collection;
+
+    my $now_dt = DateTime->now( time_zone => FixMyStreet->local_time_zone );
+
+    my $collection_date_str = $open_collection->get_extra_field_value('DATE');
+    my $collection_dt       = DateTime::Format::Strptime->new(
+        pattern   => '%FT%T',
+        time_zone => FixMyStreet->local_time_zone,
+    )->parse_datetime($collection_date_str);
+
+    return $self->_check_within_bulky_refund_window( $now_dt,
+        $collection_dt );
+}
+
+sub _check_within_bulky_refund_window {
+    my ( undef, $now_dt, $collection_dt ) = @_;
+
+    my $cutoff_dt = $collection_dt->clone->set( hour => 6, minute => 45 )
+        ->subtract( hours => 24 );
+
+    return $now_dt <= $cutoff_dt;
+}
+
+sub within_bulky_cancel_window {
+    my $self = shift;
+    my $c    = $self->{c};
+
+    my $open_collection = $c->stash->{property}{pending_bulky_collection};
+    return 0 unless $open_collection;
+
+    my $now_dt = DateTime->now( time_zone => FixMyStreet->local_time_zone );
+
+    my $collection_date_str = $open_collection->get_extra_field_value('DATE');
+    my $collection_dt       = DateTime::Format::Strptime->new(
+        pattern   => '%FT%T',
+        time_zone => FixMyStreet->local_time_zone,
+    )->parse_datetime($collection_date_str);
+
+    return $self->_check_within_bulky_cancel_window( $now_dt,
+        $collection_dt );
+}
+
+sub _check_within_bulky_cancel_window {
+    my ( undef, $now_dt, $collection_dt ) = @_;
+
+    # Collection date must be at least a day later than today's date for
```

Comment:
Unless I've missed something, I think this function can be done as:
```
my $collection_day = $collection_dt->clone->truncate( to => 'day' );
my $cutoff_dt = $collection_dt->clone->subtract( minutes => 5 );
return $now_dt < $cutoff_dt;
```

Reply:
Oh duh 🤦